wagtail/wagtail
# Remove initTooltips include & add upgrade considerations

## Overview

- Fully remove initTooltips util and replace with Controller usage, add release notes and backwards compatible support for data-tippy-* (on initial load only) to be removed fully in a future release.
- Adds more unit tests to the tooltip & dropdown controller code, specifically for the hide on esc behaviour.
- Fixes #10668 (last part of the issue), built on #10869

## Checklist

-   [x] For new features: Has the documentation been updated accordingly? **upgrade considerations only**

## How to test this change

- In the code changes, the `plugins` are now moved to the controller files with one being exported (hide on esc) as it's used by other code. These functions are 100% the same.
- A key change however, is that we no longer need to dynamically init tooltips in the chooser modal or when dynamic search results get patched into the DOM.


To validate this, do some testing as follows;

1. Load the pages search `http://localhost:8000/admin/pages/search/?q=breads` and then confirm, on initial load, that the time ago style tooltips show.
2. Now, do a search for a different page, when the HTML gets updated with the new results, confirm these new results have functional tooltips.
3. Now, edit the home page (or any page with a page chooser in use), within the chooser modal, confirm the tooltips work correctly in the initial modal results.
4. When, do a search within the modal to confirm that the tooltips still work.
